### PR TITLE
[google_maps_flutter] Mark iOS arm64 simulators as unsupported

### PR DESCRIPTION
**Pull request by jmagman**
Before https://github.com/flutter/flutter/pull/87244, any app using the `google_maps_flutter` plugin would not build on an M1 Mac.  

Replace deprecated `'VALID_ARCHS[sdk=iphonesimulator*]' => 'x86_64'` with `'EXCLUDED_ARCHS[sdk=iphonesimulator*]' => 'arm64'`.

Fixes https://github.com/flutter/flutter/issues/87242
See also https://github.com/flutter/plugins/pull/4208

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 2.1.0
```
Comment (jmagman) [inline packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md:1]: Is this worth a minor bump, or should I make it a patch 2.0.8?

**Comment by jmagman** [comment]
Test failure to be fixed in flutter/flutter: flutter/flutter#87362

File: packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md (hunk 1 above)
Comment (stuartmorgan) [reply packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md:1]: I would say 2.0.8; it seems much more like a bugfix than a feature.

**Review by stuartmorgan — APPROVED** [review]
LGTM with version nit.

File: packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md (hunk 1 above)
Comment (jmagman) [reply packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md:1]: Done.

**Pull request merged**